### PR TITLE
DM-28237: Improve ATSpectrograph timeout handling

### PR DESCRIPTION
**Pull request by tribeiro**
Update setup.cfg to reformat code with black.
Update configuration schema to add timeout configuration.
Set timeout configuration to the model class.
Implement backward compatibility with previous xml file for sending new configuration values.
If connection fails during enable, disconnect before returning.
Add `begin_enable` method to send INPROGRESS acknowledgment.

### CODE REVIEW

File: python/lsst/ts/atspectrograph/atspec_csc.py [hunk 1]
```diff
@@ -832,6 +862,13 @@ async def configure(self, config):
                 for key in gratings_str:
                     gratings_str[key] += ","
 
+        # Backward compatibility with xml < 7.2
```
Comment (patrickingraham) [inline python/lsst/ts/atspectrograph/atspec_csc.py:865]: Don't understand this but ok...
what should it be replaced with once 7.2 is deployed?

File: schema/ATSpectrograph.yaml [hunk 2]
```diff
@@ -18,6 +18,22 @@ properties:
     decription: Port for the controller
     type: integer
     default: 9999
+  connection_timeout:
+    description: >-
+        How long to wait for a response from the low level controller when
+        stablishing the connection (in seconds).
```
Comment (patrickingraham) [inline schema/ATSpectrograph.yaml:24]: establishing

File: schema/ATSpectrograph.yaml [hunk 3]
```diff
@@ -18,6 +18,22 @@ properties:
...
+    type: number
+    default: 120.
```
Comment (patrickingraham) [inline schema/ATSpectrograph.yaml:26]: this seems insanely long!

**Review by patrickingraham — APPROVED** [review]
Look at the comments. A few minor corrections should be done.

File: python/lsst/ts/atspectrograph/atspec_csc.py (hunk 1 above)
Comment (tribeiro) [reply python/lsst/ts/atspectrograph/atspec_csc.py:865]: nothing... it is a change intended for xml 7.2 but to keep the code backward compatible with xml 7.1.

**Pull request merged**